File: IdeaProject/КИТПО/kitpo_lr_4/src/main/scala/MyTreeStr.scala
class MyTreeStr(val newType: String) extends MyTree {
  _type = newType
  override def ParseValue(str: String): Any = {
    return str
  }

  override def comapareTo(obj1: Any, obj2: Any): Int = {
    if (obj1.asInstanceOf[String].compareTo(obj2.asInstanceOf[String]) < 0) return -1
    else if (obj1.asInstanceOf[String].compareTo(obj2.asInstanceOf[String]) > 0) return 1
    else return 0
  }
  override def Clone(): MyTree = {
    return new MyTreeStr("String")
  }

}

File: IdeaProject/КИТПО/kitpo_lr1/src/MyTreeStr.java
public class MyTreeStr extends MyTree  {
    MyTreeStr(String type) {
        super(type);
    }

    @Override
    public Object parseValue(String str) {
        return (str);
    }

    @Override
    public int comapareTo(Object obj1, Object obj2) {
        String str1 = (String) obj1;
        String str2 = (String) obj2;

        if(str1.length() == str2.length()) return 0;
        else if(str1.length() > str2.length()){

            return 1;
        }
        else {

            return  -1;
        }

    }
}
